lib/proxy/http_download.rb

```diff
@@ -14,11 +14,15 @@ def initialize(src, dst, read_timeout = nil, connect_timeout = nil, dns_timeout
       dns_timeout ||= DEFAULT_CONNECT_TIMEOUT
       connect_timeout ||= DEFAULT_DNS_TIMEOUT
 
-      args = [wget, "--connect-timeout=#{connect_timeout}",
+      args = ["--connect-timeout=#{connect_timeout}",
               "--dns-timeout=#{dns_timeout}",
               "--read-timeout=#{read_timeout}",
-              "--tries=3", "-nv", "-c", src.to_s, "-O", dst.to_s]
-      args << "--no-check-certificate" unless verify_server_cert
+              "--tries=3",
+              "--timestamping", # turn on timestamping to prevent redownloads
+              "--no-if-modified-since", # but use HTTP HEAD instead If-Modified-Since
+              "-nv", src.to_s, "-O", dst.to_s]
+      args.unshift("--no-check-certificate") unless verify_server_cert
+      args.unshift(wget)
```

Comment:
This makes it much more effort to review these patches. This is really the reason these things aren't merged quickly. If you would at least separate the commits then it'd be easier.

Reply:
> Took the opportunity to change behavior in this new code - instead of appending cert flag, I am prepending it. Looks cleaner to me.

In the context of cleaning up: wouldn't it be better to consistently only use long options, i.e. --no-verbose and --output-document instead of -nv and -O?

Reply:
Yes, I thought about the same thing but didn't want to drag on this PR further. IMHO it should have been 1 PR to change the behavior and one PR to change the coding style.

Reply:
I agree, this is exactly what I did in: https://github.com/theforeman/smart-proxy/pull/799/files

Generally, I tend not to do such changes, but in this case I was planning to do backport in a more simple way (just replacing the arguments in question in a new commit).